types/types.go

```diff
@@ -65,3 +66,28 @@ type Event interface {
 	Msg(msg string)
 	Msgf(format string, args ...interface{})
 }
+
+type Context interface {
+	Logger() SubLogger
```

Comment:
This API is solid, but should likely inherit from Event. To make that happen we should replace the `Event` interface's definition to be a generic`Types` interface with `Str`, `Bool`, etc. that return type `[T]`. Then have the `Event` interface inherit the `Types` interface and return type `[Event]`. 

Then, we have the `Context` interface inherit the `Types` interface with type `[Context]`

Reply:
The main reason for this is that if we add types in the future we can ahve the defined in a single location.

Reply:
We should also rename `Context` to `ContextBuilder` or something similar since you can't use `Context` on its own.

Reply:
> This API is solid, but should likely inherit from Event. To make that happen we should replace the `Event` interface's definition to be a generic`Types` interface with `Str`, `Bool`, etc. that return type `[T]`. Then have the `Event` interface inherit the `Types` interface and return type `[Event]`.
> 
> Then, we have the `Context` interface inherit the `Types` interface with type `[Context]`

Ah, makes sense. Implemented in 1a23525 (this helped me realized that I forgot to implement the `Err()` method 😅)

> We should also rename `Context` to `ContextBuilder` or something similar since you can't use `Context` on its own.

I'm not sure if `ContextBuilder` makes sense to me. I feel like this name implies that you're building a `Context` when the end-goal is to build a logger. `LoggerBuilder` also sounds strange 🤔 

While unlikely, you could use different contexts:
```go
logger := ...

baseCtx := logger.With().Str("a", "b")
otherCtx := baseCtx.Str("c", "d")

baseCtxLogger := baseCtx.Logger()
otherCtxLogger := otherCtx.Logger()
```

And if it helps as precedence, I stole this name from [zerolog](https://pkg.go.dev/github.com/rs/zerolog#Context) 😄

Reply:
Let's keep it then